docs/_frontend/src/models/SearchCombobox/SearchCombobox.js

```diff
@@ -0,0 +1,181 @@
+import classNames from 'classnames';
+import { useCombobox, useMultipleSelection } from 'downshift';
+import React, { useState } from 'react';
+import debounce from 'lodash.debounce';
+import styles from './SearchCombobox.module.css';
+
+const SearchCombobox = ({
+  label,
+  autoComplete,
+  initialSelectedItems,
+  onChange,
+}) => {
+  const [inputValue, setInputValue] = useState('');
+  const [items, setItems] = useState([]);
+  const [loading, setLoading] = useState(false);
+
+  const {
+    getSelectedItemProps,
+    getDropdownProps,
+    addSelectedItem,
+    removeSelectedItem,
+    selectedItems,
+  } = useMultipleSelection({
+    initialSelectedItems,
+    onSelectedItemsChange: ({ selectedItems }) => {
+      onChange(selectedItems);
+    },
+  });
+
+  const fetchItems = (input, selectedItem) => {
+    autoComplete(input)
+      .then((f) => {
+        if (f.ok) {
+          return f.json();
+        }
+      })
+      .then((result) => {
+        setItems(
+          result.data.filter(
+            (item) => !selectedItems.includes(item) && item !== selectedItem
+          )
+        );
+      })
+      .finally(() => setLoading(false));
+  };
+
+  const findItemsButChill = debounce(fetchItems, 350);
```

Comment:
It does not work as expected because we create the `findItemsButChill` on every render. We use memorization here.

Reply:
@pabla , Changed to used the useEffect hook since we are calling an side effect API function. Please have a look